test/data/ansible/check_ostree.yaml

```diff
@@ -71,6 +71,29 @@
       when: ignition == "true" and ((ansible_facts['distribution'] == 'RedHat' and ansible_facts['distribution_version'] is version('9.2', '>=')) or
             (ansible_facts['distribution'] == 'CentOS' and (ansible_facts['distribution_version'] == '9')))
 
+    - name: check systemd service correctly started on firstboot
+      block:
+
+        - name: check hello.service logs
+          command: journalctl -b -0 -u hello.service
+          register: result_hello_service_log
+
+        - assert:
+            that:
+              - "'Hello, World!' in result_hello_service_log.stdout"
+            fail_msg: "hello.service doesn't have the correct log"
+            success_msg: "hello.service started and working"
```

Comment:
I don't understand how is this assertion related to `systemd.condition-first-boot=true` argument ? Can you clarify ?

Reply:
FTR see https://redhat-internal.slack.com/archives/C0235DZB0DT/p1677592474917199